tdms/CMakeLists.txt

```diff
@@ -136,6 +136,11 @@ install(TARGETS tdms)
 if (BUILD_TESTING)
     enable_testing()
 
+    # find all the array.h dataype tests
+    file(GLOB ARRAY_TEST_FILES
```

Comment:
Does [GLOB_RECURSE](https://cmake.org/cmake/help/latest/command/file.html#glob-recurse) work from the `unit/` directory?

Then we don't need to maintain the long list of files...

Reply:
I mean:
```cmake
file(GLOB_RECURSE UNIT_TEST_FILES "${CMAKE_SOURCE_DIR}/tests/units/*.cpp")
```
and then
```cmake
add_executable(tdms_tests ${UNIT_TEST_FILES})
```
...? Would save us 12 lines of CMakeLists configuration.

Reply:
I was curious enough to do it myself: #160. Targets this branch.

Reply:
I would guess it does but don't quote me on that :sweat_smile:. Keep me posted on #160.

Wondering whether it's worth us thinking about breaking out into multiple test executables. Currently we just have "test all", but as we add more functionality to the code we might want to break this down into "array (unit) tests", "field (unit) tests", "interpolation (unit) tests" etc.

Reply:
Well at the moment, our units are super-fast. So I wouldn't split up unless we really need to down the road. (We will anyway need to do some CMake gymnastics when we do that so we can have more specific globs.)

And [Catch2 has some filtering capabilities](https://github.com/catchorg/Catch2/blob/devel/docs/command-line.md#specifying-which-tests-to-run):
```bash
➤ ./tdms_tests "Matrix: allocation and deallocation"
Filters: Matrix: allocation and deallocation
Randomness seeded to: 2454314993
[2022-11-02 16:38:20.995] [info] [test_Matrix.cpp:12] == Testing Matrix allocation/deallocation
===============================================================================
All tests passed (67 assertions in 1 test case)
```

Reply:
I leave it up to you though... I won't be offended if you decline #160.

Reply:
Throw #160 in - it'll also save a diff in `CMakeLists.txt` every time I add another test for all our classes!
The tests in #160 are failing because of the MacOS problem of course, but feel free to merge it into here anyway. They're not breaking anything that we have power over.